### PR TITLE
Grid filter state in url

### PR DESCRIPTION
**Pull request by vojtatranta**

### CODE REVIEW

File: wzk/ui/grid/Filter.coffee [hunk 1]
```diff
@@ -53,6 +69,16 @@ class wzk.ui.grid.Filter extends goog.events.EventTarget
     goog.dom.forms.setValue @el, val
 
   ###*
+    @protected
+    @param {string} operator
+  ###
+  setOperator: (operator) ->
```
Comment (LukasRychtecky) [inline wzk/ui/grid/Filter.coffee]: Is not necessary to use internal setter

File: wzk/ui/grid/FilterWatcher.coffee [hunk 2]
```diff
@@ -13,24 +14,53 @@ class wzk.ui.grid.FilterWatcher extends goog.events.EventTarget
...
+
       @fields[filter.getName()] = filter
       @watchField filter
 
-    @grid.listen wzk.ui.grid.Grid.EventType.LOADED, @handleLoad
+    @updateInitialState()
+
+    @query
+
+  updateInitialState: ->
```
Comment (LukasRychtecky) [inline wzk/ui/grid/FilterWatcher.coffee]: protected?

File: wzk/ui/grid/FilterWatcher.coffee [hunk 3]
```diff
@@ -103,7 +148,32 @@ class wzk.ui.grid.FilterWatcher extends goog.events.EventTarget
     @query
 
   ###*
+    @return {Object}
+  ###
+  getParams: ->
+    params = {}
+    filters = @fields
```
Comment (LukasRychtecky) [inline wzk/ui/grid/FilterWatcher.coffee]: useless local var

File: wzk/ui/grid/Paginator.coffee [hunk 4]
```diff
@@ -190,6 +190,11 @@ class wzk.ui.grid.Paginator extends wzk.ui.Component
   getBases: ->
     @bases ? @defBases
 
+  setPage: (page) ->
```
Comment (LukasRychtecky) [inline wzk/ui/grid/Paginator.coffee]: setPage: (@page) ->

and remove next lline

File: wzk/ui/grid/StateHolder.coffee [hunk 5]
```diff
@@ -48,14 +56,22 @@ class wzk.ui.grid.StateHolder
   ###
   handle: (@paginator, @watcher) ->
     @paginator.listen wzk.ui.grid.Paginator.EventType.GO_TO, @handleGoTo
-    @history.listen goog.history.EventType.NAVIGATE, @handleNav
+    @history.listen goog.history.EventType.NAVIGATE, @uriChanged
     @watcher.listen wzk.ui.grid.FilterWatcher.EventType.CHANGED, @handleFilterChanged
 
+  uriChanged: =>
```
Comment (LukasRychtecky) [inline wzk/ui/grid/StateHolder.coffee]: protected

File: wzk/ui/grid/StateHolder.coffee [hunk 6]
```diff
@@ -48,14 +56,22 @@ class wzk.ui.grid.StateHolder
...
+    @watcher.updateInitialState()
+    @dispatchChanged()
+
+  dispatchChanged: ->
```
Comment (LukasRychtecky) [inline wzk/ui/grid/StateHolder.coffee]: protected

File: wzk/ui/grid/FilterWatcher.coffee (hunk 2 above)
Comment (vojtatranta) [inline wzk/ui/grid/FilterWatcher.coffee]: Its public, it is called in StateHolder.

File: wzk/uri/Frag.coffee [hunk 7]
```diff
@@ -29,9 +29,10 @@ class wzk.uri.Frag
   ###*
     @param {string} k
     @param {*} v
+    @param {boolean=} force
```
Comment (LukasRychtecky) [inline wzk/uri/Frag.coffee]: default false

**Pull request merged**